sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.concurrent.{Callable, ConcurrentLinkedQueue, Future}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.Queue
+
+import ai.rapids.cudf.{HostMemoryBuffer, NvtxColor, NvtxRange}
+import com.nvidia.spark.rapids.GpuMetric.PEAK_DEVICE_MEMORY
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+
+import org.apache.spark.TaskContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.read.PartitionReader
+import org.apache.spark.sql.execution.datasources.PartitionedFile
+import org.apache.spark.sql.rapids.InputFileUtils
+import org.apache.spark.sql.rapids.execution.TrampolineUtil
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+/**
+ * The base HostMemoryBuffer information read from a single file.
+ */
+trait HostMemoryBuffersWithMetaDataBase {
+  // PartitionedFile to be read
+  def partitionedFile: PartitionedFile
```

Comment:
why did you change this to partitionedFile vs the 2 individual filePath and partitionValues I had?

Reply:
Seems the original HostMemoryBuffersWithMetaData has re-defined all of the fields of PartitionedFiles  https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala#L1315

Reply:
```
case class PartitionedFile(
    partitionValues: InternalRow,
    filePath: String,
    start: Long,
    length: Long,
    @transient locations: Array[String] = Array.empty) {

```

I guess the only thing left now is the locations and it looks like those stay referenced for the entire partition reader so this should be fine then.

Reply:
Hi Tom, Could you tell what does "those stay referenced for the entire partition reader " mean? Since PartitionedFile was passed to Task from Driver. so with transient annotation, locations here should be empty. am I right?